pkg/providers/docker/docker.go

```diff
@@ -531,6 +545,56 @@ func (p *provider) UpdateKubeConfig(content *[]byte, clusterName string) error {
 	}
 }
 
+// BuildKubeconfigReader creates a KubeconfigReader.
+func BuildKubeconfigReader(provider providers.Provider, unauthClient *kubernetes.UnAuthClient, timeout time.Duration, backoff time.Duration) providers.KubeconfigReader {
+	return KubeconfigReader{
+		client:   unauthClient,
+		timeout:  timeout,
+		backoff:  backoff,
+		provider: provider,
+	}
+}
+
+// WriteKubeconfig retrieves the contents of the specified cluster's kubeconfig from a secret and copies it to an io.Writer.
+func (kr KubeconfigReader) WriteKubeconfig(ctx context.Context, clusterName, kubeconfigPath string, w io.Writer) error {
+	kubeconfigSecret := &corev1.Secret{}
+
+	err := retrier.New(
+		kr.timeout,
+		retrier.WithRetryPolicy(retrier.BackOffPolicy(kr.backoff)),
+	).Retry(func() error {
+		client, err := kr.client.BuildClientFromKubeconfig(kubeconfigPath)
+		if err != nil {
+			return err
+		}
+
+		err = client.Get(ctx, fmt.Sprintf("%s-kubeconfig", clusterName), constants.EksaSystemNamespace, kubeconfigSecret)
```

Comment:
We can probably create a method in the clusterapi package, like `KubeconfigSecretName` so there is only one place that knows how to generate this name from the cluster name